### ISSUE TITLE
Define custom breakpoint for responsive calendar

### ISSUE THREAD

**Issue by NateWr** [stephenharris/Event-Organiser#338]
Hey Stephen, I'm looking to define a custom breakpoint at which the full calendar collapses into the vertical view. I've found the JS hook system you're using to filter the calendar args (nice!). However, in this particular case, I don't believe I have access to `this`.

https://github.com/stephenharris/Event-Organiser/blob/develop/js/frontend.js#L241-L247

```
                windowResize: function(view) {
                	if( view.calendar.options.responsive && $(window).width() < 514 ){
                		$(this).fullCalendar( 'changeView', _eoResponsiveViewMap[view.calendar.options.previousView] );
                	} else {
                		$(this).fullCalendar( 'changeView', view.calendar.options.previousView );
                	}
                },
```

If I were to overwrite the `windowResize` callback with my own, is there a global variable that will give me access to call `$(this).fullCalendar()`?

**Comment by stephenharris** [comment]
Doesn't `this` always be the referencing the calendar element? 

My understanding was that `this` refers to the context at invocation, not at definition. In that sense `this` doesn't exist yet: the calendar hasn't been created. But when it is, a JavaScript instance (fullCalendar instance) is created with the `windowResize` method, and what that is invoked, `this` refers to that instance.

I've not tested this but you should be able to do:

     args.windowResize = function( view ) {
          //... with this
     }

If not, I'd consider making the break point and configurable option in itself (that you can change with the filter). 

But note that there are some CSS breakpoints which match the JavaScript - so you'd need to update that too!

*3.0.0 also allows themes to 'move' the plug-in stylesheet, or map it to a different one (e.g. `style.css`). (see [Templates > Added Hooks](http://wp-event-organiser.com/blog/announcements/event-organiser-3-0-0-rc-1/)). We can chat about this at the meet-up tomorrow...*

**Comment by NateWr** [comment]
Of course you're right!

I'm going with a fully disabled CSS implementation. Took me a while to get the structure of the jQuery UI markup but going well so far. See you on ~~Wednesday~~ Thursday.

**Comment by NateWr** [comment]
Ok, so I was able to override the functions related to the responsive breakpoint, however, I had to duplicate the `_eoResponsiveViewMap` variable which is used in the `defaultView` and `windowResize` functions:

https://github.com/stephenharris/Event-Organiser/blob/develop/js/frontend.js#L76-L85

I suspect that won't be changing lots but it's always nice to minimize the potential for breaking changes in the future.

Another option would be to actually have the breakpoint be a filtered variable. That's the thing a themer is most likely to want to change (the functions will remain identical), and it'd be a more "lightweight" way to step in with less potential for conflicts down the road.
